src/Folklore/GraphQL/Support/Field.php

```diff
@@ -4,15 +4,23 @@
 
 use Illuminate\Support\Fluent;
 use Folklore\GraphQL\Error\AuthorizationError;
+use Folklore\GraphQL\Type\Definition\ResolveInfo;
 
 class Field extends Fluent
 {
 
     /**
      * Override this in your queries or mutations
      * to provide custom authorization
+     *
+     * @param $root
+     * @param array $args
+     * @param array $context
```

Comment:
How did you determine the type?

In my project it's always the authenticated $user model…

Reply:
To answer my own question: via `\Folklore\GraphQL\GraphQLController::queryContext` it's established that the context is _always_ the authenticated user (or null).

So a proper type hint can only be something like `mixed` or `object`, but definitely not `array`.

Reply:
I agree, the current type hint is incorrect. But I think it should be set to the laravel `Illuminate\Contracts\Auth\Authenticatable ` interface or `null` to improve IDE autocomplete

Reply:
No, because you can use your own `GraphQLController` and establish your own Context.

In fact, this is what I'm doing because the authenticated User is not enough context for my application. My context is a completely different kind of object which.

The only correct type is `mixed` but in your actual resolve/authorize/… functions you can use the type declaration you expect (ignoring for a second the problem this PR solved.

Reply:
In that case, you would be overriding this method anyway right so would update the typehints? There's no issue _really_ if the docblocks are wrong as well. I think it somewhat makes sense if this package maintains consistency but it doesn't really matter.

At the least, your point about `mixed` is probably best 👍